### PR TITLE
feat(query): add "EC2 Instance Using API Keys" for Terraform

### PR DESCRIPTION
**Pull request by rogeriopeixotocx**
**Proposed Changes**
- EC2 instance without associated role using AWS API keys

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/terraform/aws/ec2_instance_using_api_keys/query.rego [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+CxPolicy[result] {
+	doc := input.document[i]
+	resource := doc.resource.aws_instance[name]
+
+	remote := resource.provisioner["remote-exec"]
+	check_aws_api_keys_or_config_files(remote)
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("aws_instance[%s]", [name]),
```
Comment (joaoReigota1) [inline assets/queries/terraform/aws/ec2_instance_using_api_keys/query.rego:55]:
```suggestion
		"searchKey": sprintf("aws_instance[%s].provisioner", [name]),
```

File: assets/queries/terraform/aws/ec2_instance_using_api_keys/query.rego [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+CxPolicy[result] {
+	doc := input.document[i]
+	resource := doc.resource.aws_instance[name]
+
+	file := resource.provisioner.file
+	contains(file.destination, aws_cli_config_files[_])
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("aws_instance[%s]", [name]),
```
Comment (joaoReigota1) [inline assets/queries/terraform/aws/ec2_instance_using_api_keys/query.rego:71]:
```suggestion
		"searchKey": sprintf("aws_instance[%s].provisioner", [name]),
```

**Review by joaoReigota1 — CHANGES_REQUESTED** [review]

File: assets/queries/terraform/aws/ec2_instance_using_api_keys/query.rego [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	resource := doc.resource.aws_instance[name]
+
+	check_aws_api_keys(resource.user_data)
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("aws_instance[%s]", [name]),
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": sprintf("aws_instance[%s].user_data is being used to configure AWS API keys", [name]),
+		"keyActualValue": sprintf("aws_instance[%s] should be using iam_instance_profile to assign a role with permissions", [name]),
```
Comment (rafaela-soares) [inline assets/queries/terraform/aws/ec2_instance_using_api_keys/query.rego:26]:
```
"keyExpectedValue": sprintf("aws_instance[%s] should be using iam_instance_profile to assign a role with permissions", [name]),
"keyActualValue": sprintf("aws_instance[%s].user_data is being used to configure AWS API keys", [name]),
```

The same applies for the rules below

**Review by rafaela-soares — DISMISSED** [review]

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Review by joaoReigota1 — APPROVED** [review]
LGTM

**Pull request merged**